### PR TITLE
mongodb: 4.0.12 -> 4.2.5

### PR DESCRIPTION
**Pull request by kfiz**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/cheetah3/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ lib, buildPythonPackage, fetchurl }:
+
+buildPythonPackage rec {
+  pname = "cheetah3";
+  version = "3.2.4";
+
+  src = fetchurl {
```
Comment (FRidh) [inline pkgs/development/python-modules/cheetah3/default.nix:7]: fetchPypi

please read the Python section from the Nixpkgs manual

File: pkgs/development/python-modules/psutil/default.nix [hunk 2]
```diff
@@ -16,7 +16,7 @@ buildPythonPackage rec {
 
   # arch doesn't report frequency is the same way
   # tests segfaults on darwin https://github.com/giampaolo/psutil/issues/1715
-  doCheck = stdenv.isDarwin || stdenv.isx86_64;
+  doCheck = if stdenv.isDarwin then false else stdenv.isx86_64;
```
Comment (FRidh) [inline pkgs/development/python-modules/psutil/default.nix:19]: Why disable tests?

**Review by FRidh — CHANGES_REQUESTED** [review]
Why two Python 3 scons versions? I don't think we should do that. Have only one. I also suggest splitting the whole scons change into a separate PR.

File: pkgs/development/python-modules/psutil/default.nix (hunk 2 above)
Comment (kfiz) [reply pkgs/development/python-modules/psutil/default.nix:19]: For psutil tests don't work on darwin see: https://github.com/NixOS/nixpkgs/pull/82524

**Comment by kfiz** [comment]
> Why two Python 3 scons versions? I don't think we should do that. Have only one. I also suggest splitting the whole scons change into a separate PR.

mongodb 4.2 seems to need a scons version that was built by a python3 interpreter. The current version seems to be built by python 2.7, see [https://github.com/NixOS/nixpkgs/blob/9ec439e009098b3132848c525a033a602f74faed/pkgs/development/tools/build-managers/scons/common.nix](url)

File: pkgs/development/tools/build-managers/scons3/setup-hook.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
+sconsBuildPhase() {
```
Comment (FRidh) [inline pkgs/development/tools/build-managers/scons3/setup-hook.sh:1]: is this a copy of the existing setup hook?

Comment (kfiz) [reply pkgs/development/tools/build-managers/scons3/setup-hook.sh:1]: yes.

**Review by primeos — DISMISSED** [review]
Regarding SCons: That change for Python 3 shouldn't be required anymore as we finally merged #75877 which is now also in `master`. After a rebase you should be able to drop these SCons changes.

File: pkgs/development/python-modules/cheetah3/default.nix (hunk 1 above)
Comment (kfiz) [reply pkgs/development/python-modules/cheetah3/default.nix:7]: I can't seem to get the package via fetchPypi. Not sure as to what is the reason for that. fetchPypi tries the domains `files.pythonhosted.org` and `pypi.io` and fails with both.

**Comment by grahamc** [comment]
@GrahamcOfBorg eval

**Comment by adrian-gierakowski** [comment]
Anything specific blocking this PR? Would be nice to have it merged.

As a side note: would it be possible to get mongodb to be built by hydra and included in binary chance again? Even though the licence is not officially recognised as “free” by FSF there is nothing in it that should detract from building and redistributing the binary.

**Comment by kfiz** [comment]
I think we can close this. @pjjw bumped this to 4.2.8, which was merged 3 days ago.
See dbd0f3e957e6327c736ae030d763368efb1c5254

**Comment by jonringer** [comment]
@kfiz thanks for opening a PR, sorry it seems that your efforts were shadowed by another commit.

But I agree that this should probably be closed now that there's a newer version on master

**Pull request closed without merging**